mbed_lstools/lstools_base.py

```diff
@@ -61,10 +64,52 @@ def __init__(self, **kwargs):
         platform_dbs.append(LOCAL_PLATFORM_DATABASE)
         self.plat_db = PlatformDatabase(platform_dbs,
                                         primary_database=platform_dbs[0])
+        self.list_unmounted = list_unmounted
 
         if 'skip_retarget' not in kwargs or not kwargs['skip_retarget']:
             self.retarget()
 
+    @abstractmethod
+    def find_candidates(self):
+        """Find all candidate devices connected to this computer
+
+        Note: Should not open any files
+
+        @return A dict with the keys 'mount_point', 'serial_port' and 'target_id'
+        """
+        raise NotImplemented
+
+    def list_mbeds(self):
+        """ List details of connected devices
+        @return Returns list of structures with detailed info about each mbed
+        @details Function returns list of dictionaries with mbed attributes 'mount_point', TargetID name etc.
+        Function returns mbed list with platform names if possible
+        """
+        candidates = list(self.find_candidates())
+        logger.debug("Candidates for display %r", candidates)
+        result = []
+        for device in candidates:
+            if not device['mount_point'] :
+                if  (device['target_id'] and device['serial_port'] and
+                     not self.list_unmounted):
+                    logger.warning(
+                        "MBED with target id '%s' is connected, but not mounted. "
+                        "Use the '-u' flag to include it in the list.",
+                        device['target_id'])
+            else:
+                htm_target_id = self.read_htm_target_id(device['mount_point'])
+                if htm_target_id:
+                    logging.debug("Found htm target id, %s, for usb target id %s",
```

Comment:
I like this better, nice.

One question, the previous version also included the target_id based on the usb: https://github.com/ARMmbed/mbed-ls/blob/master/mbed_lstools/lstools_win7.py#L58

It'd probably be good to not drop this, are you ok adding this back in? Maybe the `find_candidates` calls should just assign to `target_id_usb_id` and this function here can assign the `target_id`? Since that's what functionally happens.

Reply:
Oh god, this is such small nit I'm sorry:

"Found htm target id, %s, for usb target id %s"

I think the `,` between `target id` and `%s` is unneeded.

But feel free to completely ignore this.

Reply:
Sure, that's a good idea.

Reply:
Do I have to call it `target_id_usb_id`? it's redundant to have 2 `id`s in one string. What do you think of `usb_target_id`? or `usb_id`? (everything's a target 😉 )

Reply:
For legacy sake, yes unfortunately. I agree it's redundant.